### ISSUE TITLE
Delete or undo last N messages

### ISSUE THREAD

**Issue by b4hand** [slackhq/hubot-slack#144]
It would be nice to be able to delete or undo the last message or last N messages that hubot sends.

I know this can be done manually via an admin, but would prefer a programmatic solution.

I'd be willing to accept deletion of just a single message id alternatively.

**Comment by evansolomon** [comment]
I think this will need support from the websocket API to edit messages.  Could be wrong, but can't find anything on it in the docs https://api.slack.com/rtm.  @paulhammond?

**Comment by b4hand** [comment]
I saw this https://api.slack.com/methods/chat.delete, but I wasn't sure how that related to the RTM api.

**Comment by paulhammond** [comment]
Sorry for the delay in replying!

The API token used to connect to the RTM API also has the ability to call web API methods - and the node-slack-client code used by hubot will have the ability to update and delete messages in the next version (see slackhq/node-slack-client#14 and slackhq/node-slack-client#17). So, once that's out you'll just need to call the right methods on the message object!

**Comment by brianantonelli** [comment]
I wrote a Hubot plugin for misc Slack utilities of which the first is a command for deleting the last N Hubot posts. I wanted to use the Slack Node Client as @paulhammond referenced above, but unfortunately there are some issues with their SDK in which it doesn't return a list of a channel's conversation history. Because of this issue (there are pull requests addressing it) I chose to interface directly with Slack's REST API for my module. 

See more here: https://github.com/Cox-Automotive/hubot-slack-utils
